site/components/src/main/java/uk/nhs/hee/web/components/MiniHubComponent.java

```diff
@@ -1,61 +1,69 @@
-                String guidanceName = (String) request.getRequestContext().getResolvedSiteMapItem().getLocalParameters().get("1");
-                for (int i = 0; i < guidancePages.size(); i++) {
-                    Guidance guidance = guidancePages.get(i);
-                    if (guidance.getName().equalsIgnoreCase(guidanceName)) {
-                        currentGuidance = guidance;
-                        if (i > 0) {
-                            previousGuidance = guidancePages.get(i - 1);
-                        }
-                        if (i < guidancePages.size() - 1) {
-                            nextGuidance = guidancePages.get(i + 1);
-                        }
-                        break;
-                    }
-                }
-            } else {
-                accessFromRootHub = true;
-                // There is restriction in the CMS to make sure having at least one guidance on hub
-                currentGuidance = guidancePages.get(0);
-                if (guidancePages.size() > 1) {
-                    nextGuidance = guidancePages.get(1);
-                }
+        List<MinihubSection> miniHubSections = getMiniHubSections(document);
+        request.setModel("miniHubSections", miniHubSections);
 
-                String minihubName = request.getRequestContext().getResolvedSiteMapItem().getHstSiteMapItem().getValue();
-                request.setModel("minihubName", minihubName);
-            }
+        MinihubSection currentSection = getCurrentSection(request, miniHubSections);
+        request.setModel("currentSection", currentSection);
+    }
 
-            request.setModel("previousGuidance", previousGuidance);
-            request.setModel("currentGuidance", currentGuidance);
-            request.setModel("nextGuidance", nextGuidance);
-            request.setModel("accessFromRootHub", accessFromRootHub);
+    /**
+     * Gets siblings of current document of type MinihubSection
+     *
+     * @param document the {@link HippoBean} instance.
+     * @return the {@link List <MinihubSection>} instance sorted by Display Name
+     */
+    List<MinihubSection> getMiniHubSections(HippoBean document) {
+        List<MinihubSection> miniHubSections = document.getParentBean().getChildBeans(MinihubSection.class);
+        miniHubSections.sort((section1, section2) -> section1.getDisplayName().compareTo(section2.getDisplayName()));
+        return miniHubSections;
+    }
+
+    /**
+     * Gets the current document of type MinihubSection
+     *
+     * @param request the {@link HstRequest} instance.
+     * @param miniHubSections the {@link List<MinihubSection>} instance.
+     * @return the {@link MinihubSection} instance matching the current position
+     */
+    MinihubSection getCurrentSection(final HstRequest request, List<MinihubSection> miniHubSections) {
+        MinihubSection currentSection = null;
+        if(request.getRequestContext().getContentBean().getContentType().equals("hee:MiniHub")) {
```

Comment:
We usually extract constants instead of hardcoding values.

Reply:
Couldn't find where these are defined.. noticed that you've done similar in e.g. [listing page](https://github.com/Manifesto-Digital/hee-cms-platform/blob/fc79e3946b13004d5bb4c1bb89a32bddf2a9fc52/site/components/src/main/java/uk/nhs/hee/web/components/ListingPageComponent.java#L75)?

Maybe one for wider discussion, and seperate clean up ticket to capture all?

Reply:
Sure, we can do a clean up at the end when we review the entire functionality.